PyInstaller/hooks/hook-PyQt5.QtQml.py

```diff
@@ -36,3 +38,481 @@
+
+        with open(self.analysis_file, mode='rb') as analys:
+            for line in analys:
+                if re.findall('[ ]+[#]', str(line)):
+                    # skip it is a comment in the code
+                    continue
+                else:
+                    # find line that contains a qml file
+                    for search_query in self.search_words:
+                        query = '.' + search_query + '[(].*?.*?.*?[)]'
+                        hyp_file = re.findall(query, str(line))
+
+                        if hyp_file:
+                            # we will use the first file
+                            first = hyp_file[0].replace(search_query, '')
+                            second = first[2:-1]
+                            if self.is_raw_string(second):
+                                return True
+                            else:
+                                return False
+                            break
+
+                        else:
+                            continue
+
+    def is_raw_string(self, raw):
+
+        # Here we read the line to see if there is
+        # a genuine qml filepath.
+        if re.findall('([+]|,|[)])', raw) or not re.findall('(\'|")', raw):
+            # It is a variable
+            return False
+
+        else:
+            if re.findall('qrc:', raw):  # it is a qrc
+                return False
+            else:
+                if raw[0] == '"':
+                    self.main_qml = raw.replace('"', '')
+                    return True
+
+                elif raw[0] == "'":
+                    self.main_qml = raw.replace("'", "")
+                    return True
+                else:
+                    return False
+
+    def reconstruct_qml_path(self):
+
+        # This is where we verify if the
+        # qml filepath is a relative path
+        # to the application's folder
+        # if it's not, we simply do nothing.
+        # it can't be a full path.
+        if os.path.exists(self.main_qml):
+            return False
+        else:
+            self.append_to_data(self.analysis_folder, self.main_qml)
+            self._first_handle(self.analysis_folder, self.main_qml)
+
+    def _crawl_qml_file(self, c_file):
+
+        # This is where we parse the qml file to look for
+        # import statements.
+
+        with open(c_file, mode='r', encoding='utf-8') as fh:
+            for line in fh:
+                if re.findall('[ ]+[//]', line):
+                    # skip it's a comment line
+                    continue
+                else:
+                    self.find_imports(line)
+                    # set the qml file to the current qml file
+                    # since the image path should be relative to
+                    # that file
+                    self.current_qml_file = c_file
+                    self.find_images(line)
+
+    def find_imports(self, statement):
+        import_stats = re.findall('import .*?.*?.*?$', statement)
```

Comment:
Can you please give an example, what exactly should be matched.

I assume this is not matching what you expect, esp, this would match `import asdhajkdhfjkahdfk`, too.

Also this place could be used to detect all cases you want to catch,